src/btree/bt_cursor.c

```diff
@@ -25,32 +25,25 @@ typedef struct {
  *     Determine if a given key is within the bounds set on a cursor.
  */
 static int
-__btcur_bounds_contains_key(
-  WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_ITEM *key, bool *key_out_of_boundsp, bool *upperp)
+__btcur_bounds_contains_key(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_ITEM *key,
+  uint64_t recno, bool *key_out_of_boundsp, bool *upperp)
```

Comment:
The layering here seems kind of broken to me, I'd have liked this to be a self contained call. As in callers just make a call into `btcur_bounds_contains_key`, and the function checks whether bounds are enabled, and finds the key from the cursor.

This is exacerbated by cursor_valid having explicit key/recno passed in. I did a review of the code and the same thing appears to be passed in as key/recno in all callers (except some callers check cbt->compare).

I'd like to see the layering here fixed, I know it makes this change bigger, but I think it will be simpler and more obviously correct.

Reply:
We had a follow up conversation, and Jie etc tried the solution I proposed, but it was a risky change.

I'm going to back off for now. Get this change in as-is and we can consider the cleanup later.